src/core/include/openvino/op/result.hpp

```diff
@@ -67,7 +67,8 @@ class OPENVINO_API Result : public Op {
     /// \brief Allows a value to be used as a function result.
     ///
     /// \param arg Node that produces the input tensor.
-    Result(const Output<Node>& arg);
+    /// \param use_input_names  When true Result will use input node tensor names as output names (Default: false).
+    Result(const Output<Node>& arg, bool use_input_names = false);
```

Comment:
This change is not compatible with https://github.com/openvinotoolkit/model_api/blob/master/model_api/cpp/utils/src/image_utils.cpp#L176
 error: undefined reference to 'ov::op::v0::Result::Result(ov::Output<ov::Node> const&)'
Perhaps an additional constructor would be safer to avoid a breaking change.
I wonder if the default use_input_names = false is the right choice. That would make the empty outputs when postprocessing is added.

Reply:
Added additional constructor with 2nd parameter.

By default the `use_input_tensor` should be false as tensor names should not be moved.
User can explicit set the Result names using input node names or keep them on that node and add dedicated tensor names to Result. The Result's dedicated names will stay on Result node even prepostproecessig applied

In case of `ov::Model` creation from `OutputVector` the Result nodes can be created and for these Results is better to set names based on input node tensor, is expected all outputs are model outputs.